io/write.go

```diff
@@ -0,0 +1,60 @@
+package io
+
+import (
+	"io/fs"
+	"os"
+)
+
+// WriteFileToDisk writes data to specified file with perm file mode.
+// If file exists trucates it and forces new permissions; otherwise creates it with specified permission
+// This method uses same parameters as ioutil.WriteFile but persists the newly created file on storage
+// using f.sync.
+func WriteFileToDisk(file string, fileBytes []byte, perm fs.FileMode) error {
+	f, err := createFileForWrite(file, perm)
+	if err != nil {
+		return err
+	}
+	return writeSyncClose(f, fileBytes)
+}
+
+// WriteFileTransactional does a write with sync like WriteFileToDisk
+// Additionally, if there is an error after opening,
+// it attempts to remove the file in question
+func WriteFileTransactional(file string, fileBytes []byte, perm fs.FileMode) error {
```

Comment:
This function is used in 10 places where the code was performing a remove.
If you look at the first usage, `GenCertPair`, the caller (lnd.go) does a file existence check. So an empty file could lead to a very confusing error situation where there won't be a next attempt.

Reply:
IMO that's an issue with the existence check not checking for a valid cert. Adding a remove after io is failing is a hail-mary and I'd rather not have a dupe function that isn't actually transactional. Maybe a bool can be passed in that determines whether it should be removed after failing sync?

Reply:
> IMO that's an issue with the existence check not checking for a valid cert

You might be right, but getting rid of `os.Remove` expands the scope of this PR into something that requires debates about extreme edge cases and the trade-offs in how they are handled. This PR maintains the existing behavior and just adds `fsync` according to the issue it resolves. We can get the sync behavior merged now and separately address whether we want to attempt file removal.

> isn't actually transactional

I wanted to keep this patch simple for now with the exact same behavior (just add sync), but it is possible to use the same function call signature but upgrade it to fully transactional by executing a file move. That is another reason why I went with two separate functions for the API.

> Maybe a bool can be passed in that determines whether it should be removed after failing sync?

Passing in just a bool could make for a bad api. In fact, some go linters have a rule to flag that. This is because
* it isn't obvious at the caller what true/false means
* when said bool is checked in the caller the complexity of the function increases due to branching
 
Of course, there are other approaches to API design. For example Uber advises to [convert a naked bool to a custom enum type](https://github.com/uber-go/guide/blob/master/style.md#avoid-naked-parameters).

Reply:
@Crypt-iQ okay, I made the function do a transactional rename in https://github.com/lightningnetwork/lnd/pull/6768/commits/2f1af4b11f219951c77204b2771346ee492550e7

Reply:
I think a bool is fine and the rename change isn't necessary in this PR. If you want to do a rename change, it can go in a different PR since the issue was just about changing WriteFile.